tests/integration/renderer/test_loading_states.py

```diff
@@ -167,3 +167,111 @@ def find_text(spec):
 
     find_spinners()
     find_text({1: 1, 2: 1, 3: 1, 4: 1})
+
+
+def test_update_title_default(dash_duo):
+    lock = Lock()
+
+    app = dash.Dash(__name__)
+
+    app.layout = html.Div(
+        children=[
+            html.H3("Press button see document title updating"),
+            html.Div(id="output"),
+            html.Button("Update", id="button", n_clicks=0),
+        ]
+    )
+
+    @app.callback(
+        Output("output", "children"),
+        [Input("button", "n_clicks")]
+    )
+    def update(n):
+        with lock:
+            return n
+
+    with lock:
+        dash_duo.start_server(app)
+        dash_duo.find_element("#button").click()
+        assert dash_duo.driver.title == "Updating..."
```

Comment:
Very nice tests! One thing though: your callback will execute on page load, right? So holding the lock while starting the server means we're testing the title during the initialization sequence. That's a good thing to test, but I think in at least one of these tests (and maybe all of them) we should test:
- during startup, we see the correct updating title
- when the callback is done we see the regular title
- when the callback is triggered interactively we see the updating title again

Also rather than a simple `assert`, for robustness I think it'd be better to do a `wait.until`, in case things don't happen quite in the expected order:

```py
from dash.testing.wait import until

until(lambda: dash_duo.driver.title == "Updating...", timeout=1)
```

And one last thing, if you're feeling adventurous: These test cases are nearly identical so they could be parametrized, something like:
```py
import pytest

@pytest.mark.parametrize(
    "kwargs,page_updating_title",
    [
        ({}, "Updating..."),
        ({"updating_title": None}, "Dash"),
        ({"updating_title": ""}, "Dash"),
        ({"updating_title": "Hello World"}, "Hello World")
    ]
)
def test_rdls003_update_title_default(dash_duo, updating_title, page_updating_title)
```

Reply:
I tried to implement your suggestions. I omitted an explicit test for the correct update title on startup because the callback should be called on startup afaik. I hope this is OK. The parametrize decorator is a great thing. Didn't know that until now, thanks :) Hope I did it all right with the locks. I'm not that sure about it.